source/contracts/reporting/Market.sol

```diff
@@ -170,41 +170,40 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
     }
 
     function updateTentativeWinningPayoutDistributionHash(bytes32 _payoutDistributionHash) public returns (bool) {
-        IReportingToken _reportingToken = reportingTokens[_payoutDistributionHash];
-        require(address(_reportingToken) != NULL_ADDRESS);
-
-        IReportingToken _tentativeWinningReportingToken = reportingTokens[tentativeWinningPayoutDistributionHash];
-        if (address(_tentativeWinningReportingToken) == NULL_ADDRESS) {
-            tentativeWinningPayoutDistributionHash = _payoutDistributionHash;
-            return true;
+        int256 _tentativeWinningStake = getPayoutDistributionHashStake(tentativeWinningPayoutDistributionHash);
+        int256 _secondPlaceStake = getPayoutDistributionHashStake(bestGuessSecondPlaceTentativeWinningPayoutDistributionHash);
+        int256 _payoutStake = getPayoutDistributionHashStake(_payoutDistributionHash);
+
+        if (_tentativeWinningStake >= _payoutStake && _payoutStake >= _secondPlaceStake) {
+            bestGuessSecondPlaceTentativeWinningPayoutDistributionHash = (_payoutStake > 0) ? _payoutDistributionHash : bytes32(0);
```

Comment:
I recognize why you left out the assignment to `tentativeWinningPayoutDistributionHash`, but does this create a situation where it is possible that we'll keep a `tentativeWinningPayoutDistributionHash` with a negative stake when we should be zeroing it at?

Same question (possibly different answer) for the last else and the one missing one.

Reply:
Yes it does (to both) ! I removed these before I added the ternary bit so it seemed redundant. Good catch!